site/student-lectures/cultural-algorithms/ca.py

```diff
@@ -0,0 +1,141 @@
+import math
+import random
+import numpy as np
+
+POPULATION_SIZE = 30
+GENERATIONS = 100
+ACCEPT_NUMBER = 2
+
+def fitness_function(individual):
+    return individual[0] ** 2 + individual[1] ** 2
+
+def initialize_population():
+    population = []
+    for _ in range(POPULATION_SIZE):
+        individual = [random.uniform(-10, 10) for _ in range(2)]
+        population.append(individual)
+    return population
```

Comment:
This is going to be difficult to test as you will need to mock random or something like that. 

I recommend not having this in a function and keep it at the highest level of your program